src/FormField/CheckBox.php

```diff
@@ -31,10 +44,31 @@ public function init()
         }
     }
 
+    public function set($value = null, $junk = null)
+    {
+        if (!is_bool($value)) {
+            throw new Exception(['Field\CheckBox::set() needs value to be a boolean', 'value'=>$value]);
+        }
+
+        parent::set($value);
+    }
+
     public function renderView()
     {
+        $this->template['label'] = $this->label ?: $this->caption;
+
+        if ($this->field ? $this->field->get() : $this->content) {
+            $this->template->set('checked', 'checked');
+        }
+
+        /*
+         * We don't want this displayed, because it can only affect "checked" status anyway
+         */
+        $this->content = null;
+
         $this->js(true)->checkbox();
 
+        $this->content = null; // no content again
```

Comment:
why we need to set content=null again here? can js()->checkbox() change it?

Reply:
because we output it differently. Normally for inputs the content is a valid spot, but for Radio it is not used. We keep the tag still because we can add other views in.